02_cancer_type_classification/nbconverted/lasso_range_all.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments, summary across all genes
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_valid'
+)
+
+training_dataset = 'all_other_cancers'
+results_dir = os.path.join(base_results_dir, training_dataset)
+
+metric = 'aupr'
+
+output_plots = True
+output_plots_dir = cfg.cancer_type_lasso_range_dir
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+nz_coefs_df = []
+
+# get pancancer coefs info for now
+for coef_info in au.generate_nonzero_coefficients_lasso_range(results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+nz_coefs_df = pd.DataFrame(
+    nz_coefs_df,
+    columns=['gene', 'cancer_type', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+nz_coefs_df.head()
+
+
+# ### Get performance information for each lasso penalty
+
+# In[4]:
+
+
+perf_df = au.load_prediction_results_lasso_range(results_dir, training_dataset)
+perf_df.head()
+
+
+# ### Compare feature selection with performance
+
+# In[5]:
+
+
+coefs_perf_df = (nz_coefs_df
+    .rename(columns={'cancer_type': 'holdout_cancer_type'})
+    .merge(perf_df[perf_df.signal == 'signal'],
+           on=['gene', 'holdout_cancer_type', 'seed', 'fold', 'lasso_param'])
+    .drop(columns=['signal'])
+)
+
+coefs_perf_df.head()
```

Comment:
I think it could be helpful along with this, since you can only see the rows for which experiment is `all_other_cancers`, adding in some sort of visualization of how many rows for each value of experiment there are. Does that make sense?

Reply:
There's actually only one "experiment" here at the moment, it's one of `single_cancer`, `pancancer`, or `all_other_cancers` depending on what `training_dataset` is set to at the top of the notebook. I think I used that column in a previous notebook when I was combining data from all 3 "experiments", but it's not really necessary here, at least not right now.

I'll just drop the column from this dataframe since it's not being used.